packages/storage/amplify_storage_s3_dart/lib/src/storage_s3_service/service/storage_s3_service_impl.dart

```diff
@@ -412,15 +412,19 @@ class StorageS3Service {
   }) async {
     final s3PluginOptions = options.pluginOptions as S3CopyPluginOptions? ??
         const S3CopyPluginOptions();
+    final s3ClientInfoSource =
+        getS3ClientInfo(storageBucket: options.buckets?.source);
+    final s3ClientInfoDestination =
+        getS3ClientInfo(storageBucket: options.buckets?.destination);
 
     final [sourcePath, destinationPath] = await _pathResolver.resolvePaths(
       paths: [source, destination],
     );
 
     final copyRequest = s3.CopyObjectRequest.build((builder) {
       builder
-        ..bucket = _storageOutputs.bucketName
-        ..copySource = '${_storageOutputs.bucketName}/$sourcePath'
+        ..bucket = s3ClientInfoDestination.bucketName
+        ..copySource = '${s3ClientInfoSource.bucketName}/$sourcePath'
         ..key = destinationPath
         ..metadataDirective = s3.MetadataDirective.copy;
     });
```

Comment:
Q: I think the code below this needs to be updated based on the destination bucket. For example the `_defaultS3Client` and `_storageOutputs.bucketName`. Can you double check this?

Reply:
+1